cpp/src/arrow/type.h

```diff
@@ -778,48 +791,54 @@ class ARROW_EXPORT IntervalType : public FixedWidthType {
 };
 
 // ----------------------------------------------------------------------
-// DictionaryType (for categorical or dictionary-encoded data)
+// Dictionary type (for representing categorical or dictionary-encoded
+// in memory)
 
-/// Concrete type class for dictionary data
+/// \brief Dictionary-encoded value type with data-dependent
+/// dictionary
 class ARROW_EXPORT DictionaryType : public FixedWidthType {
```

Comment:
After this refactor DictionaryType looks more like a nested type.

Reply:
It's a synthetic construct in C++ since there is no Dictionary type in the protocol metadata...